Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -133,13 +133,19 @@ class TopSitesPanel: UIViewController {
 
     private func deleteHistoryTileForSite(site: Site, atIndexPath indexPath: NSIndexPath) {
         profile.history.removeSiteFromTopSites(site) >>== {
-            self.profile.history.getTopSitesWithLimit(self.layout.thumbnailCount).uponQueue(dispatch_get_main_queue(), block: { result in
-                self.updateDataSourceWithSites(result)
-                self.deleteOrUpdateSites(result, indexPath: indexPath)
-            })
+            self.invalidateTopSites() >>== {
```

Comment:
A better name is `updateTopSitesCache`, and stacking like this makes it clearer:

```
    func reloadThumbnails() {
        self.profile.history.getTopSitesWithLimit(self.layout.thumbnailCount)
                            .uponQueue(dispatch_get_main_queue()) {
                                self.deleteOrUpdateSites(result, indexPath: indexPath)
                            }
    }

    profile.history.removeSiteFromTopSites(site)
>>> self.updateTopSitesCache
>>> reloadThumbnails
```